pkg/kubelet/kubelet.go

```diff
@@ -2276,32 +2277,73 @@ func (kl *Kubelet) rejectPod(pod *api.Pod, reason, message string) {
 		Message: "Pod " + message})
 }
 
+// getNodeAnyWay() must return a *api.Node which is required by RunGeneralPredicates().
+// The *api.Node is obtained as follows:
+// 1. When kubelet is in standaloneMode, manufactures *api.Node with kubelet
+// 2. Try to get *api.Node from kubelet's nodeInfo, if failed, we try 3.
+// 3. Try to get *api.Node from apiserver
+func (kl *Kubelet) getNodeAnyWay() (*api.Node, error) {
+	var node *api.Node
+	if kl.standaloneMode {
+		if n, err := kl.initialNodeStatus(); err == nil {
+			return n, nil
+		}
+	}
+	if n, err := kl.nodeInfo.GetNodeInfo(kl.nodeName); err == nil {
+		return n, nil
+	}
+	node, err := kl.kubeClient.Core().Nodes().Get(kl.nodeName)
```

Comment:
To be honest, I am not sure. But when `kl.nodeInfo.GetNodeInfo(kl.nodeName)` returns non-nil error, the returned node is nil and this is not acceptable for  RunGeneralPredicates(). On the other hand, `kl.kubeClient.Core().Nodes().Get(kl.nodeName)` returns the node as empty *api.Node when err != nil.

FYI: RunGeneralPredicates() uses the *api.Node only for Resource and NodeLabel checks. For other predicates, an empty *api.Node (&api.Node) is OK for RunGeneralPredicates() to make the right decision, but not nil.